src/root.component.tsx

```diff
@@ -26,15 +26,21 @@ function Root(props) {
           showPatientSummary={setShowPatientSummary}
         />
       </Route>
-      <Route
-        path="/patient/:patientUuid/chart"
-        exact
-        component={PatientChartSummary}
-      />
-      <Route
-        path="/patient/:patientUuid/chart"
-        render={routeProps => <LevelTwoRoutes match={props.match} />}
-      />
+      <main className="omrs-main-content">
+        {showPatientSummary && (
+          <Route
+            path="/patient/:patientUuid/chart"
```

Comment:
Conditionally rendering `<Routes>` is an antipattern - your routes are how your user gets to where they want to go. You do not want them to have to click on something in the UI in order for them to be able to use a route.

If the user is **exactly** on `/patient/:patientUuid/chart`, they should see the patient summary page. Additionally, the patient banner should show if the user clicks on it. Here's what I think the code should be here:

```jsx
    <BrowserRouter basename={window["getOpenmrsSpaBase"]()}>
      <Route path="/patient/:patientUuid/chart" render={routeProps => (
        <PatientBanner
          {...routeProps}
          patient={currentPatient}
          showPatientSummary={setShowPatientSummary}
         />
        )} />
      </Route>
      {/* this popover is not a route, since it's not represented by the URL. Instead, it's "transient state" that disappears when you reload the page */}
      {showPatientSummary && (
        <div className="patient-banner-popover">
          <PatientChartSummary patient={currentPatient} />
        </div>
      )}
      <main className="omrs-main-content">
        <Route
          path="/patient/:patientUuid/chart"
          exact
          component={PatientChartSummary}
        />
        <Route
          path="/patient/:patientUuid/chart"
          component={LevelTwoRoutes}
        />
      </main>
    </BrowserRouter>
```

Reply:
Hi @joeldenning,the reason why we are conditionally displaying the route is when you click on the open patient banner button, the profile shows and everything else hides no matter the route. I thought this would be hacky, but I cant think of a more elegant solution for this situation.

Reply:
That should be accomplished through z-index and modal. Not through hiding and showing the routes.